src/klein/_app.py

```diff
@@ -53,11 +80,28 @@ def _call(__klein_instance__, __klein_f__, *args, **kwargs):
 @implementer(IKleinRequest)
 class KleinRequest(object):
     def __init__(self, request):
+        # type: (Request) -> None
         self.branch_segments = [""]
-        self.mapper = None
 
-    def url_for(self, *args, **kwargs):
-        return self.mapper.build(*args, **kwargs)
+        # Don't annotate as optional, since you should never set this to None
+        self.mapper = None  # type: MapAdapter # type: ignore[assignment]
+
+    def url_for(
+        self,
+        endpoint,  # type: Text
+        values=None,  # type: Optional[Mapping[Text, Text]]
+        method=None,  # type: Optional[Text]
+        force_external=False,  # type: bool
+        append_unknown=True,  # type: bool
+    ):
+        # type: (...) -> Text
+        assert self.mapper is not None
+        return cast(
```

Comment:
Why do we know for sure that this `cast` is valid?

Reply:
Ideally, Werkzeug will add types in the future.

In the meantime, I guess we hope the [docs](https://werkzeug.palletsprojects.com/en/1.0.x/routing/?highlight=build#werkzeug.routing.MapAdapter.build) showing strings in all examples means it's returning strings all the time

Reply:
Oh wait is this a cast to *narrow*?  I thought it was a cast away from some other type, not `Any`

Reply:
If it's just to narrow I'd say you could stkip it, since the `-> Text` will fix it anyway?